### ISSUE TITLE
Player-shot arrows not picked up

### ISSUE THREAD

**Issue by SteveTownsend** [SteveTownsend/SmartHarvestSE#98]
Nexus report

> Stray arrows I've shot and that missed their target don't get picked up. Maybe intentional behavior so that arrows in flight don't get picked out of the air? But I believe Auto Harvest would pick up these arrows, but I remember that it didn't pick them up if they were floating in water.

**Comment by SteveTownsend** [comment]
Testing various cases. Back of Jorrvaskr, Whiterun on WhiteList.
Drop Ancient Nord Arrow, gets picked up:
```
J:\GitHub\SmartHarvestSE\src\Looting\tasks.cpp(1052): [VERBOSE] {43672} (2020-07-02 06:53:10 AM) Process REFR 0xff0022fe with base object Ancient Nord Arrow/0x00034182
J:\GitHub\SmartHarvestSE\src\Data\dataCase.cpp(870): [VERBOSE] {43672} (2020-07-02 06:53:10 AM) New arrow detected, save position 29796.02,-4062.92,-2728.40
J:\GitHub\SmartHarvestSE\src\Looting\tasks.cpp(1052): [VERBOSE] {43672} (2020-07-02 06:53:11 AM) Process REFR 0xff0022fe with base object Ancient Nord Arrow/0x00034182
J:\GitHub\SmartHarvestSE\src\Utilities\debugs.cpp(136): [VERBOSE] {43672} (2020-07-02 06:53:11 AM) 0x00034182 0x2a(42) [Ancient Nord Arrow] - typename ammo
J:\GitHub\SmartHarvestSE\src\Data\dataCase.cpp(888): [VERBOSE] {43672} (2020-07-02 06:53:11 AM) Arrow OK, not in flight, change in position dx=-0.27,dy=0.33,dz=-0.49
J:\GitHub\SmartHarvestSE\src\Looting\tasks.cpp(393): [VERBOSE] {43672} (2020-07-02 06:53:11 AM) SmartHarvest Ancient Nord Arrow/0x00034182 for REFR 0xff0022fe at distance 164.14 units
```
Fired Ancient Nord Arrow that wound up loose on the ground close to player (within loot radius):
```
J:\GitHub\SmartHarvestSE\src\Looting\tasks.cpp(1052): [VERBOSE] {43672} (2020-07-02 06:55:59 AM) Process REFR 0xff002301 with base object Ancient Nord Arrow/0x0003be1a
J:\GitHub\SmartHarvestSE\src\Utilities\debugs.cpp(136): [VERBOSE] {43672} (2020-07-02 06:55:59 AM) 0x0003be1a 0x32(50) [Ancient Nord Arrow] - typename ammo
J:\GitHub\SmartHarvestSE\src\FormHelpers\FormHelper.cpp(109): [VERBOSE] {43672} (2020-07-02 06:55:59 AM) Projectile 9.00
J:\GitHub\SmartHarvestSE\src\Data\dataCase.cpp(888): [VERBOSE] {43672} (2020-07-02 06:55:59 AM) Arrow OK, not in flight, change in position dx=0.00,dy=0.00,dz=0.00
J:\GitHub\SmartHarvestSE\src\Looting\tasks.cpp(393): [VERBOSE] {43672} (2020-07-02 06:55:59 AM) SmartHarvest Ancient Nord Arrow/0x0003be1a for REFR 0xff002301 at distance 126.03 units
```
Fired arrow into training dummy within loot radius
```
J:\GitHub\SmartHarvestSE\src\Looting\tasks.cpp(1052): [VERBOSE] {43672} (2020-07-02 07:00:40 AM) Process REFR 0xff0022ec with base object Ancient Nord Arrow/0x0003be1a
J:\GitHub\SmartHarvestSE\src\Utilities\debugs.cpp(136): [VERBOSE] {43672} (2020-07-02 07:00:40 AM) 0x0003be1a 0x32(50) [Ancient Nord Arrow] - typename ammo
J:\GitHub\SmartHarvestSE\src\FormHelpers\FormHelper.cpp(109): [VERBOSE] {43672} (2020-07-02 07:00:40 AM) Projectile 9.00
J:\GitHub\SmartHarvestSE\src\Data\dataCase.cpp(888): [VERBOSE] {43672} (2020-07-02 07:00:40 AM) Arrow OK, not in flight, change in position dx=0.00,dy=0.00,dz=0.00
J:\GitHub\SmartHarvestSE\src\Looting\tasks.cpp(393): [VERBOSE] {43672} (2020-07-02 07:00:40 AM) SmartHarvest Ancient Nord Arrow/0x0003be1a for REFR 0xff0022ec at distance 297.96 units
```
Fired Ancient Nord Arrow that wound up loose on the ground close to player (initially outside loot radius):
```
J:\GitHub\SmartHarvestSE\src\Looting\IRangeChecker.cpp(29): [VERBOSE] {43672} (2020-07-02 07:02:38 AM) REFR 0xff0022ec is 660.17 units away, loot range 640.00 XY-units, 448.00 Z-units
J:\GitHub\SmartHarvestSE\src\Looting\IRangeChecker.cpp(29): [VERBOSE] {43672} (2020-07-02 07:02:40 AM) REFR 0xff0022ec is 505.74 units away, loot range 640.00 XY-units, 448.00 Z-units
J:\GitHub\SmartHarvestSE\src\Looting\tasks.cpp(1052): [VERBOSE] {43672} (2020-07-02 07:02:40 AM) Process REFR 0xff0022ec with base object Ancient Nord Arrow/0x0003be1a
J:\GitHub\SmartHarvestSE\src\Data\dataCase.cpp(870): [VERBOSE] {43672} (2020-07-02 07:02:40 AM) New arrow detected, save position 29391.77,-4399.53,-2729.99
J:\GitHub\SmartHarvestSE\src\Looting\IRangeChecker.cpp(29): [VERBOSE] {43672} (2020-07-02 07:02:41 AM) REFR 0xff0022ec is 420.67 units away, loot range 640.00 XY-units, 448.00 Z-units
J:\GitHub\SmartHarvestSE\src\Looting\tasks.cpp(1052): [VERBOSE] {43672} (2020-07-02 07:02:41 AM) Process REFR 0xff0022ec with base object Ancient Nord Arrow/0x0003be1a
J:\GitHub\SmartHarvestSE\src\Utilities\debugs.cpp(136): [VERBOSE] {43672} (2020-07-02 07:02:41 AM) 0x0003be1a 0x32(50) [Ancient Nord Arrow] - typename ammo
J:\GitHub\SmartHarvestSE\src\Data\dataCase.cpp(888): [VERBOSE] {43672} (2020-07-02 07:02:41 AM) Arrow OK, not in flight, change in position dx=0.00,dy=0.00,dz=0.00
J:\GitHub\SmartHarvestSE\src\Looting\tasks.cpp(393): [VERBOSE] {43672} (2020-07-02 07:02:41 AM) SmartHarvest Ancient Nord Arrow/0x0003be1a for REFR 0xff0022ec at distance 420.67 units
```
Fired Ancient Nord Arrow into training dummy (initially outside loot radius):. Get detected but cannot autoloot - no 3D loaded
```
J:\GitHub\SmartHarvestSE\src\FormHelpers\PlayerCellHelper.cpp(58): [VERBOSE] {43672} (2020-07-02 07:11:41 AM) skip REFR, 3D not loaded Ancient Nord Arrow/0x00034182
...
J:\GitHub\SmartHarvestSE\src\FormHelpers\PlayerCellHelper.cpp(58): [VERBOSE] {43672} (2020-07-02 07:16:54 AM) skip REFR, 3D not loaded Ancient Nord Arrow/0x00034182
```
Sequence ends when manually looted. This looks like a game behaviour that I cannot resolve.

**Comment by SteveTownsend** [comment]
Releasing fixes as above in v1.6

**Comment by SteveTownsend** [comment]
Reported again on Nexus:

> SHSE still won't pick up my shot arrows from the ground or from corpses.

FormIDs : ff0011bd -> 3be17 -> _ObjectTypeUnknown_

The problem is that PROJ -> AMMO linkage is many to one, and mod logic does not properly suport that right now. Easy to repro, shoot Iron Arrows and they don't get looted:

> 2020-11-15 18:39:11.400    trace  43276 Projectile 0x0003be11/Iron Arrow has Ammo 0x0001397d/Iron Arrow
2020-11-15 18:39:11.401    trace  43276 Projectile 0x0003be11/Iron Arrow has Ammo 0x0e0c5e3e/Sevaris' Iron Arrow
2020-11-15 18:39:11.401    trace  43276 Projectile 0x0003be11/Iron Arrow has Ammo 0x0e218aec/Iron Arrow
2020-11-15 18:39:11.401    trace  43276 Projectile 0x0003be11/Iron Arrow has Ammo 0x0e2dc400/Fairy Wings
2020-11-15 18:39:11.401    trace  43276 Projectile 0x0003be11/Iron Arrow has Ammo 0x172febe6/Silver Arrow
2020-11-15 18:39:11.401    trace  43276 Projectile 0x0003be11/Iron Arrow has Ammo 0x172febe7/Khajiit Long Bow Arrow
2020-11-15 18:39:11.401    trace  43276 Projectile 0x0003be11/Iron Arrow has Ammo 0x175c4b97/Nedic Arrow
2020-11-15 18:39:11.404    trace  43276 Projectile 0x0003be11/Iron Arrow has Ammo 0xb9069070/Khajiit Long Bow Arrow
2020-11-15 18:39:11.404    trace  43276 Projectile 0x0003be11/Iron Arrow has Ammo 0xb9091ba4/Silver Arrow

> 2020-11-15 18:45:09.537    trace  37436 Process REFR 0xff001cf0 with base object Iron Arrow/0x0003be11
2020-11-15 18:45:09.537    debug  37436 Get config setting 1/1/enableharvest = 1.0
2020-11-15 18:45:09.537    trace  37436 blacklist objType == ObjectType::unknown for 0xff001cf0
2020-11-15 18:45:10.786    trace  37436 Process REFR 0xff001ca5 with base object Iron Arrow/0x0003be11
2020-11-15 18:45:10.786    debug  37436 Get config setting 1/1/enableharvest = 1.0
2020-11-15 18:45:10.786    trace  37436 blacklist objType == ObjectType::unknown for 0xff001ca5

**Comment by SteveTownsend** [comment]
Test this with _990139F7 <MASChitinArrow>_ to ensure we round trip properly (if possible) as this has PROJ _03BE19 <ArrowFalmerProjectile>_

**Comment by SteveTownsend** [comment]
Discovered a new derived class of _TESObjectREFR_ that represents _Projectile_ - this links correctly to the underlying _Ammo_ as shown by this output:
```
2020-11-26 10:04:35.857     info  45384 Projectile REFR 0xff001c95 with Base Falmer Arrow/0x0003be19 mapped to Ammo Chitin Arrow/0x990139f7
```

**Comment by SteveTownsend** [comment]
This logic now works but detection of adjacent outdoor cells needs rework

**Comment by SteveTownsend** [comment]
Adjacent cell handling fixed to check p[layer CELL distance from each CELL in the exact same worldspace, not topmost antecedent, Shot arrow pick up in Jorrvaskr back yard improved. Dropped arrows still picked up properly. Some shot arrows seem to just disappear from the world.

**Comment by SteveTownsend** [comment]
Calling this fixed for 3.7.0.11 - iterative work may be needed either for arrows or for new CELL proximity logic
